### ISSUE TITLE
Have embedded prompts configurable in private secrets

### ISSUE THREAD

**Issue by merveenoyan** [huggingface/chat-ui#346]
These days people are building products based on initial prompts and giving a task or a character to an LLM. It would be nice for Space to have a private prompt so people can build products on ChatUI docker template so people can build products based on template. @gary149 @julien-c @osanseviero

**Comment by julien-c** [comment]
in my opinion this should be done at the model repo level (using a new standard config file like `prompt_template.json`) and not at the chat-ui level, otherwise we risk reinventing the wheel

**Comment by merveenoyan** [comment]
@julien-c I feel like anything works fine so long as it can be private. Can you elaborate on model repo level though? I feel like people could still use oasst-falcon with prompt of their choice and not have their own model all the time.

**Comment by julien-c** [comment]
ah yes

well you can already set `preprompt` inside of the `MODELS` env var i think it does what you describe? cc @nsarrazin though for confirmation

**Comment by nsarrazin** [comment]
Yes if you want to customize the initial prompt then you can do it from the `MODELS` var indeed!

**Comment by merveenoyan** [comment]
Ah ok, just [saw inside the Space](https://huggingface.co/spaces/merve/my_own_oasst_falcon/blob/2b0d9236744f2521475ab3580a6c1f6b575048a2/.env.local.template#L22). Can't we bring this to UI and then add to documentation (I think this increases the awareness)?
WDYT? somewhere below @nsarrazin @julien-c  
<img width="553" alt="Screenshot 2023-07-14 at 13 12 53" src="https://github.com/huggingface/chat-ui/assets/53175384/e416aae8-b44e-43c8-b078-5ec7af1c87a5">

**Comment by nsarrazin** [comment]
We can add any vars we want in the environment variables, it wouldn't be hard to do! 😄  I'm not sure which ones we want to expose, there's a lot of options in the .env, but I agree imo at least the initial prompt would make sense. 

Does it need to be a private var or can we put it in the variables ?

**Comment by julien-c** [comment]
> Does it need to be a private var or can we put it in the variables ?

Should be a variable IMO

This could be a #good-first-issue! unless @merveenoyan wants to implem it
